zerver/webhooks/front/view.py

```diff
@@ -9,180 +9,119 @@
+    name = get_source_name(payload)
+    comment = payload['target']['data']['body']
+    return "**{name}** left a comment:\n```quote\n{comment}\n```" \
+        .format(name=name, comment=comment)
+
+def get_conversation_assigned_body(payload: Dict[Text, Any]) -> Text:
+    source_name = get_source_name(payload)
+    target_name = get_target_name(payload)
+
+    if source_name == target_name:
+        return "**{source_name}** assigned themselves." \
+            .format(source_name=source_name)
+
+    return "**{source_name}** assigned **{target_name}**." \
+        .format(source_name=source_name, target_name=target_name)
+
+def get_conversation_unassigned_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    return "Unassined by **{name}**.".format(name=name)
+
+def get_conversation_archived_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    return "Archived by **{name}**.".format(name=name)
+
+def get_conversation_reopened_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    return "Reopened by **{name}**.".format(name=name)
+
+def get_conversation_deleted_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    return "Deleted by **{name}**.".format(name=name)
+
+def get_conversation_restored_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    return "Restored by **{name}**.".format(name=name)
+
+def get_conversation_tagged_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    tag = payload['target']['data']['name']
+    return "**{name}** added tag **{tag}**.".format(name=name, tag=tag)
+
+def get_conversation_untagged_body(payload: Dict[Text, Any]) -> Text:
+    name = get_source_name(payload)
+    tag = payload['target']['data']['name']
+    return "**{name}** removed tag **{tag}**.".format(name=name, tag=tag)
+
+EVENT_FUNCTION_MAPPER = {
+    'inbound': get_inbound_message_body,
+    'outbound': get_outbound_message_body,
+    'out_reply': get_outbound_reply_body,
+    'comment': get_comment_body,
+    'mention': get_comment_body,
+    'assign': get_conversation_assigned_body,
+    'unassign': get_conversation_unassigned_body,
+    'archive': get_conversation_archived_body,
+    'reopen': get_conversation_reopened_body,
+    'trash': get_conversation_deleted_body,
+    'restore': get_conversation_restored_body,
+    'tag': get_conversation_tagged_body,
+    'untag': get_conversation_untagged_body
+}
+
+def get_body_based_on_event(event: Text) -> Any:
+    return EVENT_FUNCTION_MAPPER[event]
 
 @api_key_only_webhook_view('Front')
 @has_request_variables
 def api_front_webhook(request: HttpRequest, user_profile: UserProfile,
                       payload: Dict[Text, Any]=REQ(argument_type='body'),
                       stream: Text=REQ(default='front'),
                       topic: Optional[Text]=REQ(default='cnv_id')) -> HttpResponse:
```

Comment:
I just updated this integration to use `check_send_webhook_message` instead of `check_send_webhook_message`. I would recommend rebasing and fixing the conflicts. And after that this should be good to go! :)